src/emberCommunication.ts

```diff
@@ -25,8 +25,11 @@ export namespace Ember{
         localStorage.setItem("PB_IMPORTED_CHALLENGE", JSON.stringify(challenge))
     }
 
-    const serializedSceneToEmberScene = (scene: Scene) => { //TODO
-        return "new EscenaDuba('[O,O,O,O,O,-,-,-],[O,P,-,O,O,-,-,-],[O,-,-,O,-,-,-,-],[O,-,O,O,-,-,-,-],[O,-,O,O,O,-,-,-],[-,-,O,O,O,O,-,-],[-,-,O,O,O,O,O,O],[-,-,A,O,O,O,O,O]')"
+    export const serializedSceneToEmberScene = (scene: Scene) => { 
+        const mapToString = (map: SceneMap) => `"${JSON.stringify(map).replace(/"/g,'')}"` //[["a","a"],["b","c"]] to "[[a,a],[b,c]]"
+        const mapsAsString = scene.maps.map(mapToString).join(',')
+        
+        return `new Escena${scene.type}([${mapsAsString}])`
     }
```

Comment:
![image](https://user-images.githubusercontent.com/37090248/234090342-067a5aa7-cc23-4bc4-943e-32bbaf2296aa.png)

Reply:
Seh, el truco de todo esto es que 1) pudiste armar arriba las validaciones y tipar la estructura del mapa porque es un json plano y una lista de listas y no un string horrible que parsear y 2) no te estoy pidiendo (ni te voy a pedir) que hagas el ember to serialized :laughing: .